src/renderer/components/conversations/Timestamp.js

```diff
@@ -1,57 +1,40 @@
-const React = require('react')
-const classNames = require('classnames')
-const moment = require('moment')
-const formatRelativeTime = require('./formatRelativeTime')
+import React, { useState, useEffect } from 'react'
+import classNames from 'classnames'
+import moment from 'moment'
+import formatRelativeTime from './formatRelativeTime'
 
 const UPDATE_FREQUENCY = 60 * 1000
 
-class Timestamp extends React.Component {
-  constructor (props) {
-    super(props)
-    this.interval = null
-  }
-
-  componentDidMount () {
-    const update = () => {
-      this.setState({
-        lastUpdated: Date.now()
-      })
-    }
-    this.interval = setInterval(update, UPDATE_FREQUENCY)
-  }
-
-  componentWillUnmount () {
-    if (this.interval) {
-      clearInterval(this.interval)
-    }
-  }
-
-  render () {
-    const {
-      direction,
-      module,
-      timestamp,
-
-      extended
-    } = this.props
-    const moduleName = module || ''
-
-    if (timestamp === null || timestamp === undefined) {
-      return null
-    }
-
-    return (
-      <span
-        className={classNames(
-          moduleName,
-          direction ? `${moduleName}--${direction}` : null
-        )}
-        title={moment(timestamp).format('llll')}
-      >
-        {formatRelativeTime(timestamp, { extended })}
-      </span>
-    )
-  }
+export function useInterval (callback, delay) {
+  useEffect(() => {
+    if (delay === null || callback === null) return
+
+    const interval = setInterval(() => { callback() }, delay)
+    return () => clearInterval(interval)
+  }, [callback, delay])
 }
 
-module.exports = Timestamp
+const Timestamp = React.memo(function Timestamp (props) {
+  const { direction, module, timestamp, extended } = props
+  const moduleName = module || ''
+
+  if (timestamp === null || timestamp === undefined) return null
+
+  const calculateRelativeTime = () => formatRelativeTime(timestamp, { extended })
+  const [relativeTime, setRelativeTime] = useState(() => calculateRelativeTime())
+
+  // Update relative time every UPDATE_FREQUENCY ms
+  const recalculateRelativeTime = () => setRelativeTime(calculateRelativeTime())
+  useInterval(recalculateRelativeTime, UPDATE_FREQUENCY)
```

Comment:
where do you clear the interval again?

Reply:
The useInterval hook is taking care of this. See line 13.

Reply:
ah now I understand how it works, please add a comment that it is cleared automatically by the useHook to prevent confusion.